File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> ind;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            if (ind.count(target - nums[i]) != 0) {
                return vector<int>{i, ind[target - nums[i]]};
            }

            ind[nums[i]] = i;
        }
        return vector<int>{-1, -1};
    }
};

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for i,e := range nums {
        j, ok := m[e]

        if ok {
            return []int{i,j}
        }
        m[target - e] = i
    }
    return []int{-1,-1}
}